integrations/opennms/graph/src/main/java/org/opennms/alec/opennms/graph/AlecToOpennmsGraphConverter.java

```diff
@@ -0,0 +1,446 @@
+import org.opennms.alec.features.graph.api.Vertex;
+import org.opennms.alec.features.graph.common.EdgeTimeNormalizer;
+import org.opennms.integration.api.v1.graph.immutables.ImmutableEdge;
+import org.opennms.integration.api.v1.graph.immutables.ImmutableGraph;
+import org.opennms.integration.api.v1.graph.immutables.ImmutableVertex;
+
+import com.google.common.collect.MultimapBuilder;
+import com.google.common.collect.SetMultimap;
+
+import edu.uci.ics.jung.graph.Graph;
+import edu.uci.ics.jung.graph.util.Pair;
+
+public class AlecToOpennmsGraphConverter {
+    private final Graph<Vertex, Edge> g;
+    private final List<Situation> situations;
+
+    private final ImmutableGraph.Builder graphBuilderAlec = ImmutableGraph.builder();
+    private final ImmutableGraph.Builder graphBuilderInventory = ImmutableGraph.builder();
+    private final Map<Vertex, ImmutableVertex> alecToOnmsVertexMap = new HashMap<>();
+    private final Map<String, org.opennms.integration.api.v1.graph.Vertex> alarmIdToNode = new HashMap<>();
+    private Vertex vertexWithMostAlarms = null;
+
+    private final Map<ResourceKey, org.opennms.integration.api.v1.graph.Vertex> inventoryObjectsForAlarms = new HashMap<>();
+    private final SetMultimap<org.opennms.integration.api.v1.graph.Vertex, org.opennms.integration.api.v1.graph.Vertex> garbageCollectedAlarmInventoryEdges = MultimapBuilder.hashKeys().hashSetValues().build();
+
+    private final EdgeTimeNormalizer inventoryEdgeTimeNormalizer;
+
+    private Situation situationWithMostAlarms = null;
+    private boolean includeAlarms = true;
+    private boolean filterEmptyNodes = true;
+
+    private static final String ONMS_GRAPHML_LABEL = "label";
+    private static final String ONMS_GRAPHML_TOOLTIP_TEXT = "tooltipText";
+    private static final String ONMS_GRAPHML_ICON_KEY = "iconKey";
+    private static final String ONMS_GRAPHML_FOCUS_IDS = "focus-ids";
+    private static final String ONMS_GRAPHML_FOCUS_STRATEGY = "focus-strategy";
+
+    private static final String ONMS_FOCUS_STRATEGY_SPECIFIC = "SPECIFIC";
+
+    private static final String ONMS_ICON_SWITCH = "switch";
+    private static final String ONMS_ICON_REDUCTION_KEY = "reduction_key";
+    private static final String ONMS_ICON_INTERFACE = "interface";
+    private static final String ONMS_ICON_SITUATION = "situation";
+
+    private static final String EDGE_WEIGHT_KEY = "edgeWeight";
+    private static final String CREATED_TIMESTAMP_KEY = "createdTimestamp";
+    private static final String UPDATED_TIMESTAMP_KEY = "updatedTimestamp";
+
+    private static final String GARBAGE_COLLECTED_KEY = "garbage_collected";
+    private static final String EDGE_TYPE_KEY = "type";
+    private static final String TYPE_INVENTORY = "inventory";
+    private static final String TYPE_SITUATION = "situation";
+
+    /**
+     * TODO: Patrick discuss with Jesse:
+     * - do we need the flags filterEmptyNodes and includeAlarms? I think we can remove them...
```

Comment:
@j-white do we need the flags filterEmptyNodes and includeAlarms? I think we can remove them, correct?

Reply:
Yeah, let's remove them and just use the default behavior.

Reply:
done